pkg/universe.dagger.io/alpha/kubernetes/helm/test/test.cue

```diff
@@ -15,19 +16,52 @@ dagger.#Plan & {
 		}
 	}
 	actions: test: {
-		URL: helm.#Install & {
-			name:       "test-pgsql"
-			source:     "URL"
-			URL:        "https://charts.bitnami.com/bitnami/postgresql-11.1.12.tgz"
-			kubeconfig: client.commands.kubeconfig.stdout
+		install: {
+			URL: helm.#Install & {
+				name:       "test-pgsql"
+				source:     "URL"
+				URL:        "https://charts.bitnami.com/bitnami/postgresql-11.1.12.tgz"
+				kubeconfig: client.commands.kubeconfig.stdout
+			}
+			repository: helm.#Install & {
+				name:       "test-redis"
+				source:     "repository"
+				chart:      "redis"
+				repoName:   "bitnami"
+				repository: "https://charts.bitnami.com/bitnami"
+				kubeconfig: client.commands.kubeconfig.stdout
+			}
 		}
-		repository: helm.#Install & {
-			name:       "test-redis"
-			source:     "repository"
-			chart:      "redis"
-			repoName:   "bitnami"
-			repository: "https://charts.bitnami.com/bitnami"
-			kubeconfig: client.commands.kubeconfig.stdout
+		upgrade: {
+			repo: helm.#Upgrade & {
+				kubeconfig: client.commands.kubeconfig.stdout
+				workspace:  client.filesystem.".".read.contents
```

Comment:
The filesystem here should be `client.filesystem."./testdata".read.contents`

Reply:
Great catch! Without this change, I was getting `cycle error` when running the integration tests. This made me dig & discover:
1. We are installing the wrong version of CUE locally, in our dev environments. [`dagger` is currently pinned to a `0.4.1` fork](https://github.com/dagger/dagger/blob/v0.2.24/go.mod#L135). This has all the (many!) details behind it: https://github.com/dagger/dagger/issues/2318
2. We are not running integration tests for the Kubernetes packages since #936 (they would have caught the existing `cycle error` that you have spotted)

While https://github.com/dagger/dagger/pull/2807/commits/316ce0e6b4128bdb2d5f754c34c2352b4e80119b solves this immediate issue, the right fix is a bit more involved (restoring #936) and I intend to do it next, as a separate PR.

Once again, thanks for noticing this!